File: server/models/courses.go
package models

import (
	"gorm.io/gorm"
)

type Courses struct {
	gorm.Model
	Title    string    `json:"title"`
	Desc     string    `json:"description"`
	AuthorId int64     `json:"author_id"`
	Price    int64     `json:"price"`
	Lessons  []*Lesson `json:"lessons" gorm:"foreignKey:CourseId"`
	Quizes   []Quiz `json:"quizes" gorm:"foreignKey:CourseId"`
}


func (Courses) TableName() string {
    return "courses" // Specify the correct table name
}

File: client/src/types/Courses.ts
import {Lesson} from './Lessons'
import { Quiz } from './Quiz'
export type lessonsFormData = {
    title:string,
    content:string
}
export type Form =  {
    author_id:number,

    title:string,
    description:string,
    price:number,
    lessons:Lesson[]
}


export type CourseFormProps ={
    handleSubmit :() => void,

}

export type ListCourses = {
    ID?: number,
    CreatedAt?: string,
    UpdatedAt?: string,
    DeletedAt?: string | null,
    title: string,
    // desc: string,
    description: string,

    author_id: number,
    price: number
    lessons?:Lesson[]
    quizes?:Quiz[]
}